management/client.go

```diff
@@ -27,31 +29,49 @@ func (e *AzureError) Error() string {
 // Client provides a client to the Azure API.
 type Client struct {
 	publishSettings publishSettings
+	managementURL   string
 }
 
-//NewAnonymouseClient creates a new azure.Client with no credentials set.
+// ClientConfig provides a configuration for use by a Client
+type ClientConfig struct {
+	ManagementCert []byte
```

Comment:
since you switched PRs (something we shouldn't do :smile:) you lost my comment here. this ManagementCert is not a client configuration. It's regarding how the client is fundamentally built.

Assume you're adding OAuth support and now are you gonna put a `token string` right next to this? That would be weird... Config should be things that are shared regardless of how the client is built. Some examples: `managementURL`, `useHTTPS`, `logger`, `timeout`, `retryPolicy` etc...

So you should probably replace the NewClientFromConfig to take `managementCert []byte` parameter along with `config`.